frontend/src/utils/oauth.tsx

```diff
@@ -37,154 +36,216 @@ const generateCodeChallenge = (codeVerifier: string) => {
-
-    router.push(authUrl);
-}
+};
 
 /**
- * Request authorization code and provide PCKE code and code verifier to authentication server
+ * Exchange refresh token for new access and refresh tokens from Litus authorization server
  */
-const exchangeAuthorizationCode = async (code: string, codeVerifier: string) => {
-    // Proxy access-token retrieval through backend to avoid CORS issues
-    const frontendApiUri = process.env.NEXT_PUBLIC_FRONTEND_API_URL;
+export const refreshLitusTokens = async (refreshToken: string) => {
     const clientId = process.env.NEXT_PUBLIC_LITUS_API_KEY;
-    const frontendUri = process.env.NEXT_PUBLIC_FRONTEND_URL
+    const frontendUri = process.env.NEXT_PUBLIC_FRONTEND_URL;
 
-    if (!frontendApiUri || !clientId || !frontendUri) {
-        throw new Error("Error during authorization code exchange: missing environment variables for OAuth flow");
+    if (!clientId || !frontendUri) {
+        throw new Error("Token refresh failed: Missing LITUS_API_KEY or FRONTEND_URL.");
     }
 
-    const redirectUri = frontendUri + "/oauth/callback"
-    const tokenProxyUri = frontendApiUri + "/api/frontend/oauth/exchange-access-token"
+    const redirectUri = `${frontendUri}/oauth/callback`;
 
-    const response = await axios.post(tokenProxyUri, {
-        grant_type: 'authorization_code',
-        code,
+    return requestTokens({
+        grant_type: 'refresh_token',
+        refresh_token: refreshToken,
         client_id: clientId,
         redirect_uri: redirectUri,
-        code_verifier: codeVerifier,
     });
-
-    return response.data.access_token;
 };
 
 /**
- * Retrieve JWT from access token via backend endpoint
+ * Exchange Litus access token for JWT from backend
  */
-const exchangeAccessTokenForJWT = async (accessToken: string) => {
+const requestJWT = async (accessToken: string) => {
     const backendAuthUrl = process.env.NEXT_PUBLIC_BURGIECLAN_BACKEND_AUTH;
 
     if (!backendAuthUrl) {
-        throw new Error("Error during JWT exchange: missing environment variables for OAuth flow");
+        throw new Error("JWT request failed: BACKEND_AUTH_URL is not defined.");
     }
 
-    const response = await axios.post(backendAuthUrl, {
-        accessToken: accessToken
-    }, {
-        headers: {
-            'accept': 'application/ld+json',
-            'Content-Type': 'application/ld+json'
-        }
-    });
+    try {
+        const response = await axios.post(backendAuthUrl, { accessToken }, {
+            headers: {
+                'Accept': 'application/ld+json',
+                'Content-Type': 'application/ld+json'
+            }
+        });
+        return response.data.token;
+    } catch (error) {
+        throw new Error(`Failed to exchange access token for JWT: ${error.response?.data?.message || error.message}`);
+    }
+};
 
-    return response.data.token;
+/**
+ * Decode and parse JWT
+ */
+const parseJWT = (jwt: string): string => {
```

Comment:
return type is not a string but any because JSON.parse returns any. You could include some checks and return { exp: number, otherattribute: string, ...}